File: back/src/main/java/web/api/br/formulario/enums/StatusCadastroEnum.java
package web.api.br.formulario.enums;
import java.util.Optional;
import java.util.Arrays;
import java.util.List;

public enum StatusCadastroEnum {

    ATIVO(true, "ATIVO"),
    INATIVO(false, "INATIVO");

    private final boolean id;
    private final String descricao;

    StatusCadastroEnum(boolean id, String descricao) {
        this.id = id;
        this.descricao = descricao;
    }

    public boolean getId() {
        return id;
    }

    public String getDescricao() {
        return descricao;
    }

    public static List<StatusCadastroEnum> getAllValues() {
        return Arrays.asList(StatusCadastroEnum.values());
    }

    public static Optional<StatusCadastroEnum> getById(boolean id) {
        return Arrays.stream(StatusCadastroEnum.values())
                .filter(status -> status.id == id)
                .findFirst();
    }

    public static Optional<StatusCadastroEnum> getByDescricao(String descricao) {
        return Arrays.stream(StatusCadastroEnum.values())
                .filter(status -> status.descricao.equalsIgnoreCase(descricao))
                .findFirst();
    }
}

File: front/src/app/core/enums/status-cadastro.enum.ts
const listaStatusSolicitacao: StatusCadastroEnum[] = [];

export class StatusCadastroEnum {
  static readonly ATIVO = new StatusCadastroEnum(true, 'Ativo');
  static readonly INATIVO = new StatusCadastroEnum(false, 'Inativo');

  private constructor(
    public readonly id: boolean,
    public readonly descricao: string
  ) {
    listaStatusSolicitacao.push(this);
  }

  public static getAllValues(): StatusCadastroEnum[] {
    return listaStatusSolicitacao;
  }

  public static getById(id: boolean): StatusCadastroEnum | undefined {
    return listaStatusSolicitacao.find(
      (statusSolicitacao) => statusSolicitacao.id === id
    );
  }

  public static getByDescricao(
    descricao: string
  ): StatusCadastroEnum | undefined {
    return listaStatusSolicitacao.find(
      (statusSolicitacao) => statusSolicitacao.descricao === descricao
    );
  }
}
